File: Algorithm/Algorithm/Programmers/LEVEL1/로또의 최고순위와 최저순위.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    int best = 0;
    int worst = 0;
    for(int i = 0; i < lottos.size(); i++) {
        if(find(win_nums.begin(), win_nums.end(), lottos[i]) != win_nums.end() || lottos[i] == 0) best++;
        if(find(win_nums.begin(), win_nums.end(), lottos[i]) != win_nums.end()) worst++;
    }
    answer.push_back(best == 0  ? 6 : 7 - best);
    answer.push_back(worst == 0 ? 6 : 7 - worst);
    
    
    return answer;
}

File: Algorithm_python/프로그래머스/LEVEL 1/로또의 최고 순위와 최저 순위.py
def rank(correct):
    if correct == 0 or correct == 1:
        return 6
    else:
        return 7 - correct

def solution(lottos, win_nums):
    answer = []
    zero_count = lottos.count(0)
    correct = 0
    for num in lottos:
        for win_num in win_nums:
            if num == win_num:
                correct += 1
    answer.append(rank(correct + zero_count))
    answer.append(rank(correct))

    return answer

lottos = [44, 1, 0, 0, 31, 25]
win_nums = [31, 10, 45, 1, 6, 19]
print(solution(lottos, win_nums))